### DISCUSSION TITLE
SHGO optimiser

### DISCUSSION THREAD

**Discussion by ben18785** [pints-team/pints#1660] [New methods]
[Scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.shgo.html) has it and it seems to perform best out of their blackbox optimisers. A really good description of the method is [here](https://stefan-endres.github.io/shgo/files/shgo_slides.pdf). I think it would be really valuable to have this as a method because,

1. Think it'd useful for people to have this in an ask/tell framework,
2. It's really quite cool, mathematically, and would be good to compare with other optimisers,
3. The method not only returns the global optimum but also the various local optima.

The last point is useful because many people have proposed sampling methods for hopping between modes (see, for example, #492 #999 #964). All of these methods, however, require that the user has access to a good method for finding the modes in the first place. I think that SHGO could be really good at finding these modes as inputs to these methods.

This is mentioned in #684 but I thought it was worth including as a separate issue.

**Comment by MichaelClerx** [comment]
Big fan of this if you can find someone to do it :D
